notebooks/ClassicalSearch.ipynb

```diff
@@ -1004,6 +1015,831 @@
+    "\n",
+    "As expected, neither of these overestimates the true solution cost.\n",
+    "\n",
+    "Now, let's use Greedy Best-First Search algorithm to solve Eight Puzzle Problem. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Current State = [7, 1, 8, 0, 4, 6, 2, 3, 5]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 0  Action[name=Down]\n",
+      "\n",
+      "Current State = [7, 1, 8, 2, 4, 6, 0, 3, 5]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 1  Action[name=Right]\n",
+      "\n",
+      "Current State = [7, 1, 8, 2, 4, 6, 3, 0, 5]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 2  Action[name=Right]\n",
+      "\n",
+      "Current State = [7, 1, 8, 2, 4, 6, 3, 5, 0]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 3  Action[name=Up]\n",
+      "\n",
+      "Current State = [7, 1, 8, 2, 4, 0, 3, 5, 6]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 4  Action[name=Up]\n",
+      "\n",
+      "Current State = [7, 1, 0, 2, 4, 8, 3, 5, 6]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 5  Action[name=Left]\n",
+      "\n",
+      "Current State = [7, 0, 1, 2, 4, 8, 3, 5, 6]  Misplaced Tiles Heuristic = 7.0  isGoal = false  Path length = 6  Action[name=Left]\n",
+      "\n",
+      "Current State = [0, 7, 1, 2, 4, 8, 3, 5, 6]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 7  Action[name=Down]\n",
+      "\n",
+      "Current State = [2, 7, 1, 0, 4, 8, 3, 5, 6]  Misplaced Tiles Heuristic = 7.0  isGoal = false  Path length = 8  Action[name=Down]\n",
+      "\n",
+      "Current State = [2, 7, 1, 3, 4, 8, 0, 5, 6]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 9  Action[name=Right]\n",
+      "\n",
+      "Current State = [2, 7, 1, 3, 4, 8, 5, 0, 6]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 10  Action[name=Right]\n",
+      "\n",
+      "Current State = [2, 7, 1, 3, 4, 8, 5, 6, 0]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 11  Action[name=Up]\n",
+      "\n",
+      "Current State = [2, 7, 1, 3, 4, 0, 5, 6, 8]  Misplaced Tiles Heuristic = 5.0  isGoal = false  Path length = 12  Action[name=Up]\n",
+      "\n",
+      "Current State = [2, 7, 0, 3, 4, 1, 5, 6, 8]  Misplaced Tiles Heuristic = 5.0  isGoal = false  Path length = 13  Action[name=Left]\n",
+      "\n",
+      "Current State = [2, 0, 7, 3, 4, 1, 5, 6, 8]  Misplaced Tiles Heuristic = 5.0  isGoal = false  Path length = 14  Action[name=Left]\n",
+      "\n",
+      "Current State = [0, 2, 7, 3, 4, 1, 5, 6, 8]  Misplaced Tiles Heuristic = 4.0  isGoal = false  Path length = 15  Action[name=Down]\n",
+      "\n",
+      "Current State = [3, 2, 7, 0, 4, 1, 5, 6, 8]  Misplaced Tiles Heuristic = 6.0  isGoal = false  Path length = 16  Action[name=Down]\n",
+      "\n",
```

Comment:
Instead of printing out the result at every iteration, we should print it after skipping a few iterations, This will make the notebook compact.